Enhance finished_runs

peregrineshahin opened the pull request:
This patch intends to enhance the finished run logic to its minimal requirements.

vdbergh commented:
``` 
{"finished": True, "deleted": False}
```
I think adding `"deleted": False` makes pagination very slow since this is not in the index. I think adding it to the index was tried but did not work (Mongo's use of indexes seems very finicky). But I think it should be retried. Perhaps now it magically starts working.

peregrineshahin commented:
> 



> ```
> {"finished": True, "deleted": False}
> ```
> 
> I think adding `"deleted": False` makes pagination very slow since this is not in the index. I think adding it to the index was tried but did not work (Mongo's use of indexes seems very finicky). But I think it should be retried. Perhaps now it magically starts working.

Well if it doesn't work because of the index, is it possible to redo the index?

vdbergh commented:
> > 
> 
> > ```
> > {"finished": True, "deleted": False}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I think adding `"deleted": False` makes pagination very slow since this is not in the index. I think adding it to the index was tried but did not work (Mongo's use of indexes seems very finicky). But I think it should be retried. Perhaps now it magically starts working.
> 
> Well if it doesn't work because of the index, is it possible to redo the index?

Well that's what I wrote about...

vdbergh commented:
```
results = request.rundb.get_results(run)
```
While I think this is currently correct it seems a bit dangerous for a run that is obtained from the db. Why not use the results field directly in a run? For a finished run it will be up to date.

peregrineshahin commented:
> ```
> results = request.rundb.get_results(run)
> ```
> 
> While I think this is currently correct it seems a bit dangerous for a run that is obtained from the db. Why not use the results field directly in a run? For a finished run it will be up to date.

I pushed your suggestion.

peregrineshahin commented:
Would this still be a necessary loop?
I feel like extra operations are falling like Dominoes(?)
```
    for run in finished_runs:
        if "results_info" not in run:
            results = run["results"]
            run["results_info"] = format_results(results, run)
```

peregrineshahin commented:
Hmm I don't actually see how actually see how can the run get it's results and results_info in db in the place where it is assigned as finished :/

vdbergh commented:
> Hmm I don't actually see how actually see how can the run get it's results and results_info in db in the place where it is assigned as finished :/

I think by this code

https://github.com/glinscott/fishtest/blob/7dfa2a03676e22c789aed7e1b755b741a6bbeb57/server/fishtest/rundb.py#L1154-L1155

vdbergh commented:
> Would this still be a necessary loop? I feel like extra operations are falling like Dominoes(?)
> 
> ```
>     for run in finished_runs:
>         if "results_info" not in run:
>             results = run["results"]
>             run["results_info"] = format_results(results, run)
> ```

I think it should be the task of update_task to make sure `run["results_info"]` is correct. I think it is but I need to check.

peregrineshahin commented:
> > Hmm I don't actually see how actually see how can the run get it's results and results_info in db in the place where it is assigned as finished :/
> 
> I think by this code
> 
> https://github.com/glinscott/fishtest/blob/7dfa2a03676e22c789aed7e1b755b741a6bbeb57/server/fishtest/rundb.py#L1154-L1155

https://github.com/glinscott/fishtest/blob/a8947af75ab843be09a8da8e2a550848e980f9c1/server/fishtest/rundb.py#L1123-L1128
Okay, then I assume these lines that imply no finished run are assigned as finished with stale results

peregrineshahin commented:
> EDIT: I think it is since update_task calls stop_run (if the run is finished) and stop_run updates the results_info field.
Well if this is the case for finished runs, we might be doing it twice in L1124 and stop_run?

vdbergh commented:
> > EDIT: I think it is since update_task calls stop_run (if the run is finished) and stop_run updates the results_info field.
> 
> Well if this is the case for finished runs, we might be doing it twice in L1124 and stop_run?

Probably. But best to fix this in a separate PR I would think.

ppigazzini commented:
```
May 08 10:41:47 dfts-0 pserve[7360]:   File "/home/usr00/fishtest/server/fishtest/templates/tests.mak", line 78, in render_body
May 08 10:41:47 dfts-0 pserve[7360]:     <%include file="run_tables.mak"/>
May 08 10:41:47 dfts-0 pserve[7360]:   File "/home/usr00/fishtest/server/env/lib/python3.11/site-packages/mako/runtime.py", line 793, in _include_file
May 08 10:41:47 dfts-0 pserve[7360]:     callable_(ctx, **kwargs)
May 08 10:41:47 dfts-0 pserve[7360]:   File "/home/usr00/fishtest/server/fishtest/templates/run_tables.mak", line 30, in render_body
May 08 10:41:47 dfts-0 pserve[7360]:     <%include file="run_table.mak" args="runs=failed_runs,
May 08 10:41:47 dfts-0 pserve[7360]: TypeError: object of type 'Undefined' has no len()

```

ppigazzini commented:
```bash
git grep -n failed_runs
server/fishtest/templates/run_tables.mak:30:    <%include file="run_table.mak" args="runs=failed_runs,
server/fishtest/templates/run_tables.mak:33:                                         count=len(failed_runs),
server/fishtest/views.py:1366:    failed_runs = []
server/fishtest/views.py:1375:            failed_runs.append(run)
server/fishtest/views.py:1381:        "failed_runs": failed_runs,
```

peregrineshahin commented:
Should be alright now.

ppigazzini commented:
DEV updated, no finished runs, asking for a page of the finished runs takes few minutes with mongod at 150% CPU and then it crashes.

```yaml
May 08 11:29:01 dfts-0 pserve[12961]: Traceback (most recent call last):
May 08 11:29:01 dfts-0 pserve[12961]:   File "/home/usr00/fishtest/server/env/lib/python3.11/site-packages/pyramid_mako/__init__.py", line 148, in __call__
May 08 11:29:01 dfts-0 pserve[12961]:     result = template.render_unicode(**system)
May 08 11:29:01 dfts-0 pserve[12961]:   File "/home/usr00/fishtest/server/env/lib/python3.11/site-packages/mako/template.py", line 444, in render_unicode
May 08 11:29:01 dfts-0 pserve[12961]:     return runtime._render(
May 08 11:29:01 dfts-0 pserve[12961]:   File "/home/usr00/fishtest/server/env/lib/python3.11/site-packages/mako/runtime.py", line 874, in _render
May 08 11:29:01 dfts-0 pserve[12961]:     _render_context(
May 08 11:29:01 dfts-0 pserve[12961]:   File "/home/usr00/fishtest/server/env/lib/python3.11/site-packages/mako/runtime.py", line 916, in _render_context
May 08 11:29:01 dfts-0 pserve[12961]:     _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
May 08 11:29:01 dfts-0 pserve[12961]:   File "/home/usr00/fishtest/server/env/lib/python3.11/site-packages/mako/runtime.py", line 943, in _exec_template
May 08 11:29:01 dfts-0 pserve[12961]:     callable_(context, *args, **kwargs)
May 08 11:29:01 dfts-0 pserve[12961]:   File "/home/usr00/fishtest/server/fishtest/templates/base.mak", line 266, in render_body
May 08 11:29:01 dfts-0 pserve[12961]:     <div>${self.body()}</div>
May 08 11:29:01 dfts-0 pserve[12961]:   File "/home/usr00/fishtest/server/fishtest/templates/tests.mak", line 78, in render_body
May 08 11:29:01 dfts-0 pserve[12961]:     <%include file="run_tables.mak"/>
May 08 11:29:01 dfts-0 pserve[12961]:   File "/home/usr00/fishtest/server/env/lib/python3.11/site-packages/mako/runtime.py", line 793, in _include_file
May 08 11:29:01 dfts-0 pserve[12961]:     callable_(ctx, **kwargs)
May 08 11:29:01 dfts-0 pserve[12961]:   File "/home/usr00/fishtest/server/fishtest/templates/run_tables.mak", line 47, in render_body
May 08 11:29:01 dfts-0 pserve[12961]:     <%include file="run_table.mak" args="runs=finished_runs,
May 08 11:29:01 dfts-0 pserve[12961]:   File "/home/usr00/fishtest/server/env/lib/python3.11/site-packages/mako/runtime.py", line 793, in _include_file
May 08 11:29:01 dfts-0 pserve[12961]:     callable_(ctx, **kwargs)
May 08 11:29:01 dfts-0 pserve[12961]:   File "/home/usr00/fishtest/server/fishtest/templates/run_table.mak", line 95, in render_body
May 08 11:29:01 dfts-0 pserve[12961]:     % if not run["finished"]:
May 08 11:29:01 dfts-0 pserve[12961]: KeyError: 'finished'

```

ppigazzini commented on server/fishtest/rundb.py:
```diff
@@ -548,15 +550,18 @@ def get_finished_runs(
         )
 
-        count = self.runs.count_documents(q)
+        q_count = dict(q)
```
We want to get the count without the deleted runs, we were not able to get it in a fast way so we took the total count.

peregrineshahin replied:
Okay makes sense, I was wondering why this count is with deleted which doesn't make any sense

ppigazzini commented on server/fishtest/views.py:
```diff
@@ -1364,20 +1365,16 @@ def get_paginated_finished_runs(request):
-
-        # Look for failed runs
-        if "failed" in run and run["failed"]:
+    if page_idx == 0:
```
reminder: fix the indentation of this block (4 spaces)

ppigazzini commented:
Home page without finished runs, page=2 built in 1 minute with `mongod` at 100% CPU, logs clean.

vdbergh commented:
Personally I am not so happy with disabling the reference to the earliest tests in the pagination (since access to it in master is fast).

peregrineshahin commented:
> Personally I am not so happy with disabling the reference to the earliest tests in the pagination (since access to it in master is fast).

No This wasn't the intended behavior, there is still something wrong in the code.

ppigazzini commented:
Same behavior: very slow home with empty finished runs and mongod at 45% CPU, page=3 built in one minute with mongod at 70% CPU, logs clean. I stop testing for this afternoon.

peregrineshahin commented:
> Same behavior: very slow home with empty finished runs and mongod at 45% CPU, page=3 built in one minute with mongod at 70% CPU, logs clean. I stop testing for this afternoon.

Ok thanks, will look into it later.
But I don't understand what 'built' means here, did it build with results?

vdbergh commented:
> Same behavior: very slow home with empty finished runs and mongod at 45% CPU, page=3 built in one minute with mongod at 70% CPU, logs clean. I stop testing for this afternoon.

I am not actually seeing this. The pagination is fast even for very late pages (accessed via page=xxx in the URL).

peregrineshahin commented:
> > Same behavior: very slow home with empty finished runs and mongod at 45% CPU, page=3 built in one minute with mongod at 70% CPU, logs clean. I stop testing for this afternoon.
> 
> I am not actually seeing this. The pagination is fast even for very late pages (accessed via page=xxx in the URL).

He reverted

ppigazzini commented:
>But I don't understand what 'built' means here, did it build with results?
Rendered.

peregrineshahin commented:
Okay rendered mean this is still bugged which is good news, as opposed to it being actually slow

ppigazzini commented on server/fishtest/rundb.py:
```diff
@@ -548,15 +550,17 @@ def get_finished_runs(
 
-        # Don't show runs that were deleted
-        runs_list = [run for run in c if not run.get("deleted")]
+        count = self.runs.count_documents(q)
```
return [list(runs_list), count]

ppigazzini commented on server/fishtest/rundb.py:
```diff
@@ -548,15 +550,17 @@ def get_finished_runs(
         )
 
-        count = self.runs.count_documents(q)
+        runs_list = [run for run in c]
```
more simple:
```python
return [list(runs_list), count]
```

ppigazzini commented:
timing master, home page and the first 5 pages:
```diff
diff --git a/server/fishtest/rundb.py b/server/fishtest/rundb.py
index 8926f08..b611bef 100644
--- a/server/fishtest/rundb.py
+++ b/server/fishtest/rundb.py
@@ -552,11 +552,14 @@ class RunDb:
         c = self.runs.find(
             q, skip=skip, limit=limit, sort=[("last_updated", DESCENDING)]
         )
-
+        start = time.time()
         count = self.runs.count_documents(q)
+        print("count:", time.time() - start, flush=True)

         # Don't show runs that were deleted
+        start = time.time()
         runs_list = [run for run in c if not run.get("deleted")]
+        print("runs_list:", time.time() - start, flush=True)
         return [runs_list, count]

     def get_results(self, run, save_run=True):
```

```yaml
May 08 15:06:39 dfts-0 pserve[32572]: Starting server in PID 32572.
May 08 15:06:47 dfts-0 pserve[32572]: count: 0.058951377868652344
May 08 15:06:47 dfts-0 pserve[32572]: runs_list: 0.14502978324890137
May 08 15:06:52 dfts-0 pserve[32572]: count: 0.05503416061401367
May 08 15:06:52 dfts-0 pserve[32572]: runs_list: 0.055086374282836914
May 08 15:06:54 dfts-0 pserve[32572]: count: 0.060930490493774414
May 08 15:06:54 dfts-0 pserve[32572]: runs_list: 0.05944705009460449
May 08 15:06:56 dfts-0 pserve[32572]: count: 0.07278251647949219
May 08 15:06:56 dfts-0 pserve[32572]: runs_list: 0.06661534309387207
May 08 15:07:33 dfts-0 pserve[32572]: count: 0.055108070373535156
May 08 15:07:33 dfts-0 pserve[32572]: runs_list: 0.21022915840148926
May 08 15:07:36 dfts-0 pserve[32572]: count: 0.0805208683013916
May 08 15:07:36 dfts-0 pserve[32572]: runs_list: 0.07940316200256348
May 08 15:07:37 dfts-0 pserve[32572]: count: 0.06372809410095215
May 08 15:07:37 dfts-0 pserve[32572]: runs_list: 0.05127739906311035
May 08 15:07:38 dfts-0 pserve[32572]: count: 0.059346914291381836
May 08 15:07:38 dfts-0 pserve[32572]: runs_list: 0.14896321296691895
May 08 15:07:40 dfts-0 pserve[32572]: count: 0.055483102798461914
May 08 15:07:40 dfts-0 pserve[32572]: runs_list: 0.05922389030456543
May 08 15:07:43 dfts-0 pserve[32572]: count: 0.06847405433654785
May 08 15:07:43 dfts-0 pserve[32572]: runs_list: 0.06534194946289062
```

ppigazzini commented:
timing PR, home page and first pages.
```diff
diff --git a/server/fishtest/rundb.py b/server/fishtest/rundb.py
index c252b8a..920da63 100644
--- a/server/fishtest/rundb.py
+++ b/server/fishtest/rundb.py
@@ -558,8 +558,16 @@ class RunDb:
             projection=projection
         )

+        start = time.time()
         count = self.runs.count_documents(q)
-        return [list(runs_cursor), count]
+        print("count:", time.time() - start, flush=True)
+
+        start = time.time()
+        runs_list = list(runs_cursor)
+        print("runs_list:", time.time() - start, flush=True)
+
+        #return [list(runs_cursor), count]
+        return [runs_list, count]

     def get_results(self, run, save_run=True):
         if not run["results_stale"]:
```

```yaml
May 08 15:16:13 dfts-0 systemd[1]: Started Fishtest Server port 6544.
May 08 15:16:15 dfts-0 pserve[1257]: Starting server in PID 1257.
May 08 15:17:46 dfts-0 pserve[1257]: count: 78.03816366195679
May 08 15:17:46 dfts-0 pserve[1257]: runs_list: 0.04495835304260254
May 08 15:19:41 dfts-0 pserve[1257]: count: 82.60482740402222
May 08 15:19:41 dfts-0 pserve[1257]: runs_list: 0.08448433876037598
May 08 15:21:27 dfts-0 pserve[1257]: count: 101.1190402507782
May 08 15:21:27 dfts-0 pserve[1257]: runs_list: 0.08047175407409668
```

peregrineshahin commented on server/fishtest/rundb.py:
```diff
@@ -548,15 +550,15 @@ def get_finished_runs(
-        # Don't show runs that were deleted
-        runs_list = [run for run in c if not run.get("deleted")]
+        runs_list = list(runs_cursor)
+        count = len(runs_list)
```
I don't get how returning the count is 100x slower than master, is it because of the indexing?

ppigazzini commented:
Current indexes on `runs`:< collection
```yaml
{ '_id_': {'key': [('_id', 1)], 'v': 2},
  'finished_green_runs': { 'key': [('finished', 1), ('is_green', -1), ('last_updated', -1)],
                           'partialFilterExpression': SON([('finished', True), ('is_green', True)]),
                           'v': 2},
  'finished_ltc_runs': { 'key': [('finished', 1), ('last_updated', -1), ('tc_base', -1)],
                         'partialFilterExpression': SON([('finished', True), ('tc_base', SON([('$gte', 40)]))]),
                         'v': 2},
  'finished_runs': { 'key': [('finished', 1), ('last_updated', -1)],
                     'partialFilterExpression': SON([('finished', True)]),
                     'v': 2},
  'finished_user_runs': { 'key': [('args.username', -1), ('finished', 1), ('last_updated', -1)],
                          'partialFilterExpression': SON([('finished', True)]),
                          'v': 2},
  'finished_yellow_runs': { 'key': [('finished', 1), ('is_yellow', -1), ('last_updated', -1)],
                            'partialFilterExpression': SON([('finished', True), ('is_yellow', True)]),
                            'v': 2},
  'unfinished_runs': { 'key': [('finished', 1)],
                       'partialFilterExpression': SON([('finished', False)]),
                       'v': 2},
  'user_runs': {'key': [('args.username', -1), ('last_updated', -1)], 'v': 2}}


```

peregrineshahin commented:
> Current indexes on `runs` collection:
> 
> ```yaml
> { '_id_': {'key': [('_id', 1)], 'v': 2},
>   'finished_green_runs': { 'key': [('finished', 1), ('is_green', -1), ('last_updated', -1)],
>                            'partialFilterExpression': SON([('finished', True), ('is_green', True)]),
>                            'v': 2},
>   'finished_ltc_runs': { 'key': [('finished', 1), ('last_updated', -1), ('tc_base', -1)],
>                          'partialFilterExpression': SON([('finished', True), ('tc_base', SON([('$gte', 40)]))]),
>                          'v': 2},
>   'finished_runs': { 'key': [('finished', 1), ('last_updated', -1)],
>                      'partialFilterExpression': SON([('finished', True)]),
>                      'v': 2},
>   'finished_user_runs': { 'key': [('args.username', -1), ('finished', 1), ('last_updated', -1)],
>                           'partialFilterExpression': SON([('finished', True)]),
>                           'v': 2},
>   'finished_yellow_runs': { 'key': [('finished', 1), ('is_yellow', -1), ('last_updated', -1)],
>                             'partialFilterExpression': SON([('finished', True), ('is_yellow', True)]),
>                             'v': 2},
>   'unfinished_runs': { 'key': [('finished', 1)],
>                        'partialFilterExpression': SON([('finished', False)]),
>                        'v': 2},
>   'user_runs': {'key': [('args.username', -1), ('last_updated', -1)], 'v': 2}}
> ```

There is no way we can beat that when the index is only finished we need to add that on deleted also.

peregrineshahin commented:
have we tried this before?
```
db.collection.createIndex({ finished: 1, deleted: 1, last_updated: -1 }, { partialFilterExpression: { finished: true, deleted: false } })
```

peregrineshahin commented:
deleted runs are never showed in the whole website to begin with (except from event log), this IMO should be the way forward.

ppigazzini commented:
The home page is fast with the new index, the page=N are still slow

```yaml
-- Logs begin at Mon 2023-05-08 13:50:25 CEST, end at Mon 2023-05-08 17:49:20 CEST. --
May 08 17:41:17 dfts-0 pserve[5558]: count: 0.0865941047668457
May 08 17:41:17 dfts-0 pserve[5558]: runs_list: 0.012347221374511719
May 08 17:42:12 dfts-0 pserve[5558]: count: 0.07577729225158691
May 08 17:42:13 dfts-0 pserve[5558]: runs_list: 0.016711711883544922
May 08 17:42:27 dfts-0 pserve[5558]: count: 0.047730207443237305
May 08 17:42:27 dfts-0 pserve[5558]: runs_list: 0.012450456619262695
May 08 17:42:33 dfts-0 pserve[5558]: count: 68.63782739639282
May 08 17:42:33 dfts-0 pserve[5558]: runs_list: 0.020566463470458984
May 08 17:42:34 dfts-0 pserve[5558]: count: 0.06219482421875
May 08 17:42:34 dfts-0 pserve[5558]: runs_list: 0.015530824661254883
May 08 17:42:37 dfts-0 pserve[5558]: count: 0.08187341690063477
May 08 17:42:37 dfts-0 pserve[5558]: runs_list: 0.010805606842041016
May 08 17:44:11 dfts-0 pserve[5558]: count: 85.75881814956665
May 08 17:44:11 dfts-0 pserve[5558]: runs_list: 0.06450581550598145
May 08 17:46:49 dfts-0 pserve[5558]: count: 94.02026796340942
May 08 17:46:49 dfts-0 pserve[5558]: runs_list: 0.043329477310180664
```

Current indexes on runs:
```yaml
{ '_id_': {'key': [('_id', 1)], 'v': 2},
  'finished_deleted_runs': { 'key': [('finished', 1), ('deleted', 1), ('last_updated', -1)],
                             'partialFilterExpression': SON([('finished', True), ('deleted', False)]),
                             'v': 2},
  'finished_green_runs': { 'key': [('finished', 1), ('is_green', -1), ('last_updated', -1)],
                           'partialFilterExpression': SON([('finished', True), ('is_green', True)]),
                           'v': 2},
  'finished_ltc_runs': { 'key': [('finished', 1), ('last_updated', -1), ('tc_base', -1)],
                         'partialFilterExpression': SON([('finished', True), ('tc_base', SON([('$gte', 40)]))]),
                         'v': 2},
  'finished_runs': { 'key': [('finished', 1), ('last_updated', -1)],
                     'partialFilterExpression': SON([('finished', True)]),
                     'v': 2},
  'finished_user_runs': { 'key': [('args.username', -1), ('finished', 1), ('last_updated', -1)],
                          'partialFilterExpression': SON([('finished', True)]),
                          'v': 2},
  'finished_yellow_runs': { 'key': [('finished', 1), ('is_yellow', -1), ('last_updated', -1)],
                            'partialFilterExpression': SON([('finished', True), ('is_yellow', True)]),
                            'v': 2},
  'unfinished_runs': { 'key': [('finished', 1)],
                       'partialFilterExpression': SON([('finished', False)]),
                       'v': 2},
  'user_runs': {'key': [('args.username', -1), ('last_updated', -1)], 'v': 2}}
```

vdbergh commented:
Indexes in Mongodb appear to be a gigantic time sink.

peregrineshahin commented:
Okay now I see why the current logic on master is as is, looping in python inside the returned 25 elements is, of course, more efficient than going these thousands of records, I assume that is master when we remove a deleted run from 25 elements to only return 24 elements which user will not notice.. so this is the trick.. if I count the finished runs in home page as of now I will find 24 instead of 25 which is okay..
Now what I should do is revert the query conditions but keep other enhancements found.. no need for the new index The N page is explainable it has failed:"false" as an extra condition breaking the index.

ppigazzini commented:
DEV updated, no time to view the result, I'm leaving the office rn.

peregrineshahin commented:
I think this is okay to be merged right now.
The projection should skip some sizable data and the loop isn't necessary to obtain failed_runs other than first page, an per the discussion with vdbergh I think removing get_results function call is safe here.

ppigazzini commented:
For reference these are the only runs missing the "results_info" field, they were "deleted" before being "approved".
```c
    _id: ObjectId("6259b1c324eace8d0ab7de96"),
    _id: ObjectId("626f785e59e9c431e0b10bf6"),
    _id: ObjectId("6270bc701bb47fe1c49fed47"),
    _id: ObjectId("628c092172775f382300e394"),
    _id: ObjectId("62d6e5ac47ae1768b34a3c20"),
    _id: ObjectId("633868a835f43d649ff60092"),
    _id: ObjectId("6338809735f43d649ff603b8"),
    _id: ObjectId("6375eed29bbfe1734c448452"),
    _id: ObjectId("63b97150cd3db0c8d399b889"),
    _id: ObjectId("64018baf2644b62c3392b299"),
    _id: ObjectId("642b774c77ff3301150d6e7c"),
    _id: ObjectId("64320bbc31feee5c6d305a82"),
```

ppigazzini commented:
We can drop this
```python
        if "results_info" not in run:
            run["results_info"] = format_results(results, run)
```
that is used only in `elo_results.mak` in:
https://github.com/glinscott/fishtest/blob/b03ec11552ca268b72bcd7dc94a463edab5c373a/server/fishtest/templates/run_table.mak#L113
https://github.com/glinscott/fishtest/blob/b03ec11552ca268b72bcd7dc94a463edab5c373a/server/fishtest/templates/tests_view.mak#L56
because we dont' show the deleted runs:
https://github.com/glinscott/fishtest/blob/b03ec11552ca268b72bcd7dc94a463edab5c373a/server/fishtest/rundb.py#L560-L562

ppigazzini reviewed (approved):
LG on DEV

ppigazzini commented:
PROD updated, thank you @peregrineshahin :)
